transforms/custom_cube_models.py

```diff
@@ -0,0 +1,35 @@
+"""Add a keyvalue to prop_weighted_cube to automatically handle custom models."""
+
+from srctools import conv_int
+
+from hammeraddons.bsp_transform import trans, Context
+
+@trans('Easy Custom Cube Models')
+def custom_cube_models(ctx: Context) -> None:
+    """Add a keyvalue to prop_weighted_cube to automatically handle custom models."""
+    for ent in ctx.vmf.by_class['prop_weighted_cube']:
+        model_type = conv_int(ent['comp_custom_model_type'])
+
+        if model_type == 0: # none
+            continue
+        elif model_type == 1: # script override
+            cube_model = ent['model']
+            # Make a prop_dynamic_override to precache the model
+            ctx.vmf.create_ent(
```

Comment:
You can have this transform produce a `comp_precache_model`, since that one is set to have a `priority=1000` to run after everything else.

Reply:
Ah okay, I wasn't sure if those would run in the right order